src/Shared/DomainEventWrapperCollection.cs

```diff
@@ -1,26 +1,35 @@
 ﻿using System.Collections;
+using System.Runtime.InteropServices.ComTypes;
 
 namespace Workleap.DomainEventPropagation;
 
 internal sealed class DomainEventWrapperCollection : IReadOnlyCollection<DomainEventWrapper>
 {
     private readonly DomainEventWrapper[] _domainEventWrappers;
 
-    public DomainEventWrapperCollection(IEnumerable<DomainEventWrapper> domainEventWrappers, string domainEventName)
+    public DomainEventWrapperCollection(IEnumerable<DomainEventWrapper> domainEventWrappers, string domainEventName, EventSchema schema)
     {
         this._domainEventWrappers = domainEventWrappers.ToArray();
         this.DomainEventName = domainEventName;
+        this.DomainSchema = schema;
     }
 
     public int Count => this._domainEventWrappers.Length;
 
     public string DomainEventName { get; }
 
+    public EventSchema DomainSchema { get; }
+
     public static DomainEventWrapperCollection Create<T>(IEnumerable<T> domainEvents)
-        where T : IDomainEvent
+        where T : IDomainEvent, new()
     {
-        var domainEventWrappers = domainEvents.Select(DomainEventWrapper.Wrap);
-        return new DomainEventWrapperCollection(domainEventWrappers, DomainEventNameCache.GetName<T>());
+        var domainEventWrappers = domainEvents.Select(DomainEventWrapper.Wrap).ToArray();
+        if (domainEventWrappers.Select(x => x.DomainEventName).Distinct().Count() > 1)
```

Comment:
Si jamais tu passe une liste de type IDomainEvent pour être coquin et mettre différent types concrets dedans ça va peter dans le `DomainEventNameCache` so je crois qu'on est déjà covered et c'est pas nécessaire ici

Reply:
As discussed, it was not necessary; it was removed in #117